logging/net_appender_test.go

```diff
@@ -130,51 +130,3 @@ func TestNetLoggerBatchWrites(t *testing.T) {
 		test.That(t, server.service.logs[i].Message, test.ShouldEqual, "Some-info")
 	}
 }
-
-func TestNetLoggerBatchFailureAndRetry(t *testing.T) {
-	server := makeServerForRobotLogger(t)
-	defer server.stop()
-
-	netAppender, err := NewNetAppender(server.cloudConfig)
-	test.That(t, err, test.ShouldBeNil)
-	logger := NewDebugLogger("test logger")
-	// The stdout appender is not necessary for test correctness. But it does provide information in
-	// the output w.r.t the injected grpc errors.
-	logger.AddAppender(netAppender)
-
-	// This test will first log 10 "Some-info" logs. Followed by a single "New info" log.
-	numLogs := 11
-
-	// Inject a failure into the server handling `Log` requests.
-	server.service.logsMu.Lock()
-	server.service.logFailForSizeCount = numLogs
-	server.service.logsMu.Unlock()
-
-	for i := 0; i < numLogs-1; i++ {
-		logger.Info("Some-info")
-	}
-
-	// This test requires at least three syncs for the logs to be guaranteed received by the
-	// server. Once the log queue is full of size ten batches, the first sync will decrement
-	// `logFailForSizeCount` to 1 and return an error. The second will decrement it to a negative
-	// value and return an error. The third will succeed.
```

Comment:
This test is a good example of the behavior we're losing with this change: we have no ability to store logs that failed to `write` anymore. I don't have a great sense of how often `write`s fail, but I can imagine it's not an uncommon occurrence (especially with robots with intermittent internet access). It doesn't seem great that we can just drop whole batches of logs now, and I might need someone to convince me that's ok (or tell me that this was not even the change you had in mind)...

Reply:
I'm going to merge this change for now and think about a more elegant solution where we can reintroduce similar behavior to the one in this test.